cla_public/static-src/javascripts/modules/form-errors.js

```diff
@@ -143,18 +144,30 @@
         if (labelField.parents(".cla-currency-by-frequency").length) {
           labelField = labelField.parents(".cla-currency-by-frequency").children("legend");
 
-          $this.find(' .govuk-input').filter(function () {
-            return ($(this).val() && Number($(this).val()) > 0)
+          var inputElementWithValidEntry = $this.find(' .govuk-input').filter(function () {
+            return currencyValidation.isValidAmount($(this).val())
           })
-          .removeClass("govuk-input--error");
+
+          if(!inputElementWithValidEntry.length) {
+            $this.find(' .govuk-input')
+            .focus(function() {
+              $(".laa-currency-prefix--error")
+              .removeClass("laa-currency-prefix--error")
+            })
```

Comment:
This needs to be more specific.  It is removing the red borders for other error fields on the page, as well as itself.